File: backend/app/schemas/approval.py
import uuid
from datetime import datetime
from decimal import Decimal

from app.schemas.base import CamelModel


class ApprovalRequestResponse(CamelModel):
    id: uuid.UUID
    policy_id: uuid.UUID | None
    entity_type: str
    entity_id: uuid.UUID
    company_id: uuid.UUID
    project_id: uuid.UUID | None
    module: str
    requested_by: uuid.UUID
    assigned_to: uuid.UUID | None
    assigned_role: str | None
    status: str
    priority: str
    amount: Decimal | None
    comment: str | None
    decided_by: uuid.UUID | None
    decided_at: datetime | None
    created_at: datetime

File: frontend/src/types/approval.ts
export type ApprovalStatus = 'PENDING' | 'APPROVED' | 'REJECTED' | 'CANCELLED'
export type ApprovalPriority = 'LOW' | 'NORMAL' | 'HIGH'

export interface ApprovalRequestEntry {
  id: string
  policyId: string | null
  entityType: string
  entityId: string
  companyId: string
  projectId: string | null
  module: string
  requestedBy: string
  assignedTo: string | null
  assignedRole: string | null
  status: ApprovalStatus
  priority: ApprovalPriority
  amount: string | null
  comment: string | null
  decidedBy: string | null
  decidedAt: string | null
  createdAt: string
}
